indeplot/indeplot
# [dev] Input and plot component

## Description
from Issue #93. 

- I created a component that allows the user input for the *_Line Chart_*.  
- *Chart title and label*:  The user may enter a _chart title_ and a _label_ that describes what they are tracking.  
- *Label Type*:  Right now, the user chooses from a limited listing of _label types_  for their _labels_ (X coordinates).  These include _months of the year_, _days of the week_ or _intervals_.  
- *Add Coordinate*: In the associated drop-down, the user will select their _label_(X coordinate), then type in their _y-coordinate_ (needs to be a number or the user will receive an alert.  The user's coordinates will appear on the _Line Graph_.  Coordinates will always be added to the end.
- *Delete Coordinate*: The user may select from an existing set of coordinates from a dropdown and delete a selected coordinate.


## Additions/Removals/Changes

- I created a folder /CoordinateInput and index.js to manage the inputs
- I created a folder /hooks and created input-hook.js to manage the state updates from the inputs.
- I made significant revisions to *LineChart.js* and minor revisions to *HomePage.js*, *Chart.js*, and *ChartSelector.js* as pertains to the _Line Chart_ only.


## How to test this PR

1. Open branch locally and clear any necessary caches.
<img width="150" alt="Screen Shot 2020-11-01 at 5 11 30 PM" src="https://user-images.githubusercontent.com/34319929/97892093-93fbfa00-1cfd-11eb-9591-7010c8f0cdb7.png">

2. From the dropdown at the top, select _Line Chart_.  You should see the _Line Chart_ appear with the new _About Your Chart_ panel below and no data.
<img width="150" alt="Screen Shot 2020-11-01 at 5 11 21 PM" src="https://user-images.githubusercontent.com/34319929/97892095-93fbfa00-1cfd-11eb-88f7-26d659bb4cd9.png">

3.  Add *Test Title* in the _Title_ input.  Add *tracking rainfall* in the _Label_ input.  Click or tap Submit.  You should see the titles on the chart.
<img width="150" alt="Screen Shot 2020-11-02 at 11 25 47 AM" src="https://user-images.githubusercontent.com/34319929/97892611-387e3c00-1cfe-11eb-92ba-cc96dd40c623.png">

4.  Click or tap the _Label type_ dropdown and select *Months of the year*.  Click or tap Add Label. 
<img width="150" alt="Screen Shot 2020-11-01 at 5 12 00 PM" src="https://user-images.githubusercontent.com/34319929/97892091-93fbfa00-1cfd-11eb-97fd-d5632e3bccf0.png">

5.  Click or tap the _Labels_ dropdown and you should see _Jan_ through _Dec_.  Select *Jan*.  Tab, tab or click on _Y_ and enter *20*.  Repeat this process by choosing the following combinations:  *Feb*/ *100*, *Mar*/*300, *Apr*/*50, *Mar*/*300*.  Upon entering the last coordinate, *Mar*/*300*, please note that you are presented with an alert.  At this point, you are not allowed to add duplicates.
<img width="150" alt="Screen Shot 2020-11-01 at 5 12 40 PM" src="https://user-images.githubusercontent.com/34319929/97892090-93fbfa00-1cfd-11eb-98d4-b081e896b026.png">

6.  Tap or click on the _Color Picker_ and choose another color.  The graph should refresh with the new color.
7.  Tap or click the _Coordinate_ list and select a coordinate.  Tap or click _Delete Coordinate_.  Note that the coordinate has disappeared from the graph.
<img width="150" alt="Screen Shot 2020-11-01 at 5 14 41 PM" src="https://user-images.githubusercontent.com/34319929/97892088-93636380-1cfd-11eb-83f8-555621afe9d7.png">
8.  Select a different graph from the drop down.  Note that, at this point, the data you entered has disappeared.  The randomly generated data function produces data for this graph. 










## Type of change

Please delete options that are not relevant.

- [ X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] This PR is made on develop branch
- [X] I have mentioned above the issue number which i have solved
- [X] I have not changed anything in the files .all-contributorsrc and The Contributor section in Readme.md
- [X] My code follows the proper indentation style of the project
- [X ] I have performed a self-review of my own code

## Reviewer: [Ayan Biswas](https://github.com/ayan-biswas0412)
